hadoop-hdds/common/src/main/java/com/google/protobuf/XmlToMarkdown.java

```diff
@@ -0,0 +1,176 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.google.protobuf;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Represents a property in the XML file.
+ */
+class Property {
+  private final String name;
+  private final String value;
+  private final String tag;
+  private final String description;
+
+  Property(String name, String value, String tag, String description) {
+    this.name = name;
+    this.value = value;
+    this.tag = tag;
+    this.description = description;
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  public String getValue() {
+    return value;
+  }
+
+  public String getTag() {
+    return tag;
+  }
+
+  public String getDescription() {
+    return description;
+  }
+}
+
+/**
+ * Utility class for converting XML properties from ozone-default.xml and
+ * ozone-default-generated.xml into Markdown format. The converted content
+ * is saved to Configurations.md.
+ */
+public final class XmlToMarkdown {
+
+  private XmlToMarkdown() {
+    // Prevent instantiation
+  }
+
+  public static void main(String[] args) {
+    try {
+      // List of XML files to process
+      String[] xmlFiles = {
+          "hadoop-hdds/common/src/main/resources/ozone-default.xml",
+          "hadoop-hdds/config/target/test-classes/ozone-default-generated.xml"
+      };
+
+      List<Property> properties = new ArrayList<>();
+
+      // Load and parse each XML file
+      for (String xmlFile : xmlFiles) {
+        File inputFile = new File(xmlFile);
+        if (!inputFile.exists()) {
+          System.out.println("File not found: " + xmlFile);
+          continue;
+        }
+
+        DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
```

Comment:
Please use `XMLUtils.newSecureDocumentBuilderFactory()`.

And there is no need to get new instances of the factory or builder for each XML file.